src/URIs.jl

```diff
@@ -380,6 +380,45 @@ function splitpath(p::AbstractString)
     return elems
 end
 
+"""
+    URIs.normpath(url)
+
+Normalize the path portion of a URI by removing dot segments.
+
+Refer to:
+* https://tools.ietf.org/html/rfc3986#section-5.2.4
+"""
+normpath(url::URI) =
+    URI(scheme=url.scheme, userinfo=url.userinfo, host=url.host, port=url.port,
+        path=normpath(url.path), query=url.query, fragment=url.fragment)
+
+function normpath(p::AbstractString)
+    if isempty(p) || p == "/"
+        return p
+    elseif p == "." || p == ".."
+        return "/"
+    end
+    buf = String[]
+    for part in splitpath(p)
```

Comment:
`Base.splitpath` is in julia 1.1 only.
(or is there a `splitpath` on HTTP master that I am not aware of)
Current bounds on this package is julia 0.7

Am I right in saying
`Base.splitpath` has different functionality on windows than linux?
And that will make this wrong on windows?
https://github.com/JuliaLang/julia/pull/28156/files#diff-e0580027daf242717c633ff9cf06a750R139

Reply:
None of the path-related names from Base are imported/extended in the URIs submodule. In this case, `splitpath` refers to the function defined above this one.

Reply:
Ah, it was hidden in the fold, and is in master but not in the release branch I am currently on.